docs/source-fabric/advanced/model_init.rst

```diff
@@ -79,6 +79,17 @@ When training distributed models with :doc:`FSDP/TP <model_parallel/index>` or D
     optimizer = torch.optim.Adam(model.parameters())
     optimizer = fabric.setup_optimizers(optimizer)
 
+With DeepSpeed Stage 3, the use of :meth:`~lightning.fabric.fabric.Fabric.init_module` context manager is necessesary for the model to be sharded correctly instead of attempted to be put on the GPU in its entirety. Deepspeed requires the models and optimizer to be set up jointly.
```

Comment:
The first sentence, not wrong but it's not entirely different from FSDP and therefore the code above was already appropriate. 

I think what you wanted to mention was the second sentence "Deepspeed requires the models and optimizer to be set up jointly", right? Can we do it so we don't repeat ourselves, e.g., just a comment about calling ` model, optimizer = fabric.setup(model, optimizer)` for DeepSpeed?

Reply:
totally. i wrote that sentence by rewriting your words from an issue linked above, but i agree that in general it applies to fsdp as well. 

just to be clear i understood you - old block of code with a text comment re: deepspeed below? or still two blocks of code? or one but with two ways (one commented out?)?

Reply:
I think one code block if possible, with a code comment or text comment outside the block would be perfect. Great idea thanks